test/test.rb

```diff
@@ -178,6 +178,24 @@ def test_should_report_size_correctly_for_ios
     end
   end
 
+  def test_should_report_type_correctly_for_readed_ios
```

Comment:
Looks good, just one minor thing, can I kindly ask you to update the name of the tests like test_should_report_type_correctly_for_ios_that_have_been_read (rather than readed_ios).

Reply:
@sdsykes fixed